В магазине электроники раз в месяц проводится распродажа. Из всех товаров выбирают K товаров с самой большой ценой и делают на них скидку в 20%. По заданной информации о цене каждого из товаров и количестве товаров, на которые будет скидка, определите цену самого дорогого товара, не участвующего в распродаже, а также целую часть от суммы всех скидок.<br><br>
<b>Входные и выходные данные.</b> В первой строке входного файла находятся два числа,записанные через пробел: N – общее количество цен (натуральное число, не превышающее 10 000) и K – количество товаров со скидкой. В следующих N строках находятся значения цены каждого из товаров (все числа натуральные, не превышающие 10 000), каждое в отдельной строке. Запишите в ответе два числа: сначала цену самого дорогого товара, не участвующего в распродаже, а затем целую часть от суммы всех скидок.
Пример входного файла:<br>
10 3<br>
1800<br>
3600<br>
3700<br>
800<br>
2600<br>
2500<br>
1800<br>
1500<br>
1900<br>
1200<br>
При таких исходных данных ответ должен содержать два числа – 2500 и 1980. Пояснение: скидка будет на товары стоимостью 3700, 3600, 2600. Тогда самый дорогой товар без скидки стоит 2500, а сумма скидок 740+720+520 = 1980. 

In [ ]:
f = open("text")
line = f.readline()
first = line.split() # делим первую строку на список элементов, разделенных пробелом 
# два элемента - количество цен, количество товаров со скидкой. По умолчанию это 
# пробел, но можно указать любой символ, по которому он бы делил, например, 
# line.split("-")
n = int(first[0])
k = int(first[1])

Тот же код можно сделать короче с помощью функции map()
Первым аргументом она принимает функцию, а вторым аргументом принимает итерируемый объект, к элементом которого применяет эту функцию. (Так же можно передать ей третий, четвертый и т.д. итерируемый объект, но чаще обходятся одним)

In [3]:
def multiply(arg):
    arg = arg*10
    return arg

l = [2, 15, 32, 6]

l = map(multiply, l)
print(l)
for i in l:
    print(i)

20
150
320
60


In [4]:
f = open("26-1_умск.txt")
line = f.readline()
n, k = map(int, line.split())
print(n, k)
price = []
for i in f:
    price.append(int(i))
price.sort(reverse=True) # сортирует по убыванию
first = price[k] # первое наибольшее число, на которое не будет скидки. 
# к-ый элемент, т.к. отсчет в списке начианается с нуля
second = 0
"""Осталось вывести сумму всех скидок (не цены товара после скидки, а значения 
самих скидок):"""
for i in range(k): # k не включительно, поэтому до k
    second += price[i]*0.2
print(first, int(second))

1000 100
9000 190680


In [24]:
j = 8
for i in range(31, -1, -1):
    j+=1
    print(f'{i}*2 + {j}*3 = {i*2 + j*3}')

31*2 + 9*3 = 89
30*2 + 10*3 = 90
29*2 + 11*3 = 91
28*2 + 12*3 = 92
27*2 + 13*3 = 93
26*2 + 14*3 = 94
25*2 + 15*3 = 95
24*2 + 16*3 = 96
23*2 + 17*3 = 97
22*2 + 18*3 = 98
21*2 + 19*3 = 99
20*2 + 20*3 = 100
19*2 + 21*3 = 101
18*2 + 22*3 = 102
17*2 + 23*3 = 103
16*2 + 24*3 = 104
15*2 + 25*3 = 105
14*2 + 26*3 = 106
13*2 + 27*3 = 107
12*2 + 28*3 = 108
11*2 + 29*3 = 109
10*2 + 30*3 = 110
9*2 + 31*3 = 111
8*2 + 32*3 = 112
7*2 + 33*3 = 113
6*2 + 34*3 = 114
5*2 + 35*3 = 115
4*2 + 36*3 = 116
3*2 + 37*3 = 117
2*2 + 38*3 = 118
1*2 + 39*3 = 119
0*2 + 40*3 = 120


По итогам проведения олимпиады по программированию каждый участник получил
определённое количество баллов, различное для каждого ученика. По регламенту олимпиады победителя присуждают K лучших участников, а призёра присуждают M лучших участников, следующих за ними. По заданной информации о результатах каждого из участников определите минимальный балл призёра и минимальный балл победителя данной олимпиады.<br>
<b>Входные и выходные данные</b>: В первой строке входного файла находятся три числа, записанные через пробел: N – общее количество результатов учащихся (натуральное число, не превышающее 10 000), K – количество победителей, M – количество призёров. В следующих N строках находятся значения каждого из результатов (все числа натуральные, не превышающие 1000), каждое в отдельной строке. Запишите в ответе два числа: сначала минимальный балл призёра, а затем минимальный балл победителя данной олимпиады без пробела. Прикрепи не только краткий ответ, но и код, чтобы куратор мог его проверить.
Пример входного файла:<br>
10 2 4<br>
244<br>
39<br>
213<br>
108<br>
132<br>
18<br>
46<br>
52<br>
242<br>
179<br>
При таких исходных данных ответ должен содержать 2 числа – 108 и 242. Пояснение: победители набрали 244 и 242 балла, призёры набрали 213, 179, 132, 108 баллов. Тогда минимальный балл призёра 108, а минимальный балл победителя – 242

In [41]:
f = open('26_5_умск.txt')
line = f.readline()
n, k, m = map(int, line.split())
scores = []
for i in f:
    scores.append(int(i))
scores.sort(reverse=True)
min_win_score = scores[k-1]
min_awardee_score = scores[k + m-1]
print(min_awardee_score, min_win_score, sep="")

519909


In [49]:
a = [2, 4, 8]
print(a[0:3])

[2, 4, 8]


In [1]:
f = open("26-3 (1)_умск.txt")
line = f.readline()
n, k, m = map(int, line.split())
all_prices = []
for price in f:
    print(price)
    all_prices.append(int(price))
all_prices.sort(reverse=True)
low_premium = all_prices[m-1] # т.к. отсчет с нуля, то до m-1
budget_sum = 0
count = 0
for i in range(m, len(all_prices)):
    budget_sum += all_prices[i]
    count +=1
budget_sum = budget_sum/count
print(low_premium, int(budget_sum))

9400

7300

21400

9600

10600

8600

26500

26300

6100

28500

12800

25600

21900

20000

28400

25200

14400

28600

19200

20800

13000

7000

26100

6200

11300

18000

19600

18300

12500

11600

8300

14700

21500

24600

15600

19100

5800

10000

6300

18600

12000

22500

16500

8200

28100

15200

10900

14300

17800

6600

23400

21600

15100

6400

23600

10200

18200

22200

22400

21100

16900

29600

11900

12700

24900

27700

22800

27200

29500

7600

19300

5300

15900

9100

28700

8000

7800

14100

19900

5000

20100

16700

23900

7700

13300

29000

28200

15700

8800

10300

8100

23700

12300

6000

11000

8900

14000

6800

17000

17200

27700 15021


In [27]:
a=[100000]*10000
with open('26-3 (1)_умск.txt') as f:
    n, k, m = [int(x) for x in next(f).split()]
    s=0
    for i in range(n):
        a[i]=int(f.readline())
for i in range(n-1):
    for j in range(n - i):
        if a[j] > a[j + 1]:
            temp = a[j]
            a[j] = a[j + 1]
            a[j + 1] = temp
for i in range(k):
    s=s+a[i]
sr=int(s/k)
print(a[n-m], sr)

[9400, 7300, 21400, 9600, 10600, 8600, 26500, 26300, 6100, 28500, 12800, 25600, 21900, 20000, 28400, 25200, 14400, 28600, 19200, 20800, 13000, 7000, 26100, 6200, 11300, 18000, 19600, 18300, 12500, 11600, 8300, 14700, 21500, 24600, 15600, 19100, 5800, 10000, 6300, 18600, 12000, 22500, 16500, 8200, 28100, 15200, 10900, 14300, 17800, 6600, 23400, 21600, 15100, 6400, 23600, 10200, 18200, 22200, 22400, 21100, 16900, 29600, 11900, 12700, 24900, 27700, 22800, 27200, 29500, 7600, 19300, 5300, 15900, 9100, 28700, 8000, 7800, 14100, 19900, 5000, 20100, 16700, 23900, 7700, 13300, 29000, 28200, 15700, 8800, 10300, 8100, 23700, 12300, 6000, 11000, 8900, 14000, 6800, 17000, 17200, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 100000, 1000

In [17]:
l = [i for i in range(9, 24, 2)]
print(l)
summ=0
for i in l:
    print(f"{summ}+{i}={summ+i}")
    summ+=i
print(f"сумма всех чисел: {summ}, длина списка:{len(l)}")

[9, 11, 13, 15, 17, 19, 21, 23]
0+9=9
9+11=20
20+13=33
33+15=48
48+17=65
65+19=84
84+21=105
105+23=128
сумма всех чисел: 128, длина списка:8


In [31]:
def divis(n, m):
    return n%m==0

def expression(x, a):
    val = divis(x,a) or (not divis(x, 15)) or (not divis(x, 18))
    return val

for a in range(1, 1000):
    flag = True
    for x in range(1, 1000):
        if not expression(x, a):
            flag = False
            break
    if flag==True:
        print(a)

1
2
3
5
6
9
10
15
18
30
45
90


In [24]:
def sys_admin(file):
    f = open(file)
    users, capacity = map(int, f.readline().split())
    all_weights = []
    for i in f:
        all_weights.append(int(i))
    all_weights.sort()
    print(all_weights, capacity)
    for i in range(len(all_weights)):
        if capacity > 0:
            capacity-=all_weights[i]
        else:
            last_index = all_weights[i]
            print(last_index)
            break
sys_admin('26-2-3_sysadmin.txt')

[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 12, 12, 12, 12, 12, 12, 12, 12, 12, 13, 13, 13, 13, 13, 13, 13, 13, 13, 13, 13, 13, 13, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 14, 15, 15, 15, 15, 15, 15, 15, 15, 15, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 18, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 20, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 22, 22, 22, 22, 22, 22, 22